Empire/Empire.cpp

```diff
@@ -143,13 +144,13 @@ namespace {
             if (excess_here < EPSILON)
                 continue;
             // Transfer excess to stockpile
-            new_contributions += excess_here * stockpile_transfer_efficiency;
-            TraceLogger() << "allocated here: " << allocated_here
-                          << "  excess here: " << excess_here
+            new_contributions += excess_here;
+            TraceLogger() << "...allocated in group: " << allocated_here
+                          << "  excess in group: " << excess_here
                           << "  to stockpile: " << new_contributions;
         }
-        TraceLogger() << "starting_stockpile: " << starting_stockpile 
-                      << "  stockpile_used: " << stockpile_used;
+        if (new_contributions > stockpile_limit)
```

Comment:
Instead of this the cut off should happen in before the trace logger in line 148 or the trace statement " to stockpile:" in line 150 should move behind or into the if block.

Reply:
I made it the way it is so that the excess in each group can be seen independently. A separate log line could be added for the limit-restricted contributions if needed.

Reply:
Ah, i missed the accumulation in  new_contributions.  Forget my comment.